implementations/elixir/ockam/ockam/lib/ockam/transport/tcp/client/gen_tcp.ex

```diff
@@ -0,0 +1,44 @@
+defmodule Ockam.Transport.TCP.Client.GenTcp do
+  @moduledoc false
+
+  use Ockam.Worker
+
+  @doc false
+  @impl true
+  def setup(options, state) do
+    ip = Keyword.get_lazy(options, :ip, &default_ip/0)
+    port = Keyword.get_lazy(options, :port, &default_port/0)
+    tcp_connect_options = [:binary, active: :once]
+
+    state = Map.put(state, :ip, ip)
+    state = Map.put(state, :port, port)
+    state = Map.put(state, :tcp_connect_options, tcp_connect_options)
```

Comment:
```suggestion
    state = Map.merge(state, %{
      ip: ip,
      port: port,
      tcp_connect_options: tcp_connect_options,
    })
```

unsure about precise syntax here, but this might grow in the future and could be better handled in a single assignment.

Reply:
possibly, although I think the more likely case is that individual option validation will move to separate functions.